astrobase/periodbase/tls.py

```diff
@@ -0,0 +1,380 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# tls.py - Luke Bouma (luke@astro.princeton.edu) - Apr 2019
+
+"""
+Contains the Hippke & Heller (2019) transit-least-squared period-search
+algorithm implementation for periodbase. This depends on the external package
+written by Hippke & Heller, https://github.com/hippke/tls.
+"""
+
+#############
+## LOGGING ##
+#############
+
+import logging
+from astrobase import log_sub, log_fmt, log_date_fmt
+
+DEBUG = False
+if DEBUG:
+    level = logging.DEBUG
+else:
+    level = logging.INFO
+LOGGER = logging.getLogger(__name__)
+logging.basicConfig(
+    level=level,
+    style=log_sub,
+    format=log_fmt,
+    datefmt=log_date_fmt,
+)
+
+LOGDEBUG = LOGGER.debug
+LOGINFO = LOGGER.info
+LOGWARNING = LOGGER.warning
+LOGERROR = LOGGER.error
+LOGEXCEPTION = LOGGER.exception
+
+
+#############
+## IMPORTS ##
+#############
+
+import numpy as np
+
+from multiprocessing import Pool, cpu_count
+
+from math import fmod
+
+from numpy import (
+    nan as npnan, arange as nparange, array as nparray,
+    isfinite as npisfinite, argmax as npargmax, linspace as nplinspace,
+    ceil as npceil, argsort as npargsort, concatenate as npconcatenate
+)
+
+from astropy import units as u
+
+try:
+    from transitleastsquares import transitleastsquares
+except:
+    errmsg = (
+        'tried importing transitleastsquares and failed.\n'
+        'are you sure you have installed it correctly?\n'
+        'see https://transitleastsquares.readthedocs.io/en/latest/Installation.html'
+    )
+    raise AssertionError(errmsg)
+
+
+###################
+## LOCAL IMPORTS ##
+###################
+
+from ..lcmath import sigclip_magseries
+
+
+############
+## CONFIG ##
+############
+
+NCPUS = cpu_count()
+
+
+#######################
+## UTILITY FUNCTIONS ##
+#######################
+
+def tls_parallel_pfind(times, mags, errs=None,
```

Comment:
Better to keep the `errs` as an actual arg instead of a kwarg for consistency. Can handle it being None by adding in fake errors like you do in the tests.